NuRadioMC/EvtGen/generator_skeleton.py

```diff
@@ -30,37 +24,91 @@ def generate_my_events(filename, n_events):
     # first set the meta attributes
     attributes = {}
     n_events = int(n_events)
-    attributes['n_events'] = n_events
+    attributes['n_events'] = n_events  # the number of events contained in this file
     attributes['start_event_id'] = 0
+    # define the fiducial simulation volume. Instead of specifying fiducial_rmin and fiducial_rmin one can also specify
+    # fiducial_xmin, fiducial_xmax, fiducial_ymin and fiducial_ymax
     attributes['fiducial_rmin'] = 0
     attributes['fiducial_rmax'] = 1 * units.km
     attributes['fiducial_zmin'] = 0 * units.m
     attributes['fiducial_zmax'] = -2 * units.km
+    # define the full simulation volume. Instead of specifying rmin and rmin one can also specify
+    # xmin, xmax, ymin and ymax
     attributes['rmin'] = 0
     attributes['rmax'] = 1 * units.km
     attributes['zmin'] = 0 * units.m
     attributes['zmax'] = -2 * units.km
+
+    attributes['volume'] = attributes['rmax'] ** 2 * np.pi * np.abs(attributes['zmax'])
+
+    # if only interactions on a surface (e.g. for muons from air showers) are generated, the surface area needs to be
+    # specified attributes['area']
+
+    # define the minumum and maximum energy
     attributes['Emin'] = 1 * units.EeV
     attributes['Emax'] = 1 * units.EeV
 
+    # the interval of zenith directions
+    attributes['thetamin'] = 0
+    attributes['thetamax'] = np.pi
+    # the interval of azimuths directions
+    attributes['phimin'] = 0
+    attributes['phimax'] = 2 * np.pi
+
     # now generate the events and fill all required data sets
     # here we fill all data sets with dummy values
+
+    # each line/entry specified a particle shower of certain energy.
+    # In principle only the shower direction (zeniths, azimuths fields), the shower position (xx, yy, zz fields)
+    # the shower energy, the shower type and the event group id are required. We set them first
     data_sets = {}
+    # the direction of the shower
     data_sets["azimuths"] = np.ones(n_events)
     data_sets["zeniths"] = np.ones(n_events)
+    # the position of the shower
     data_sets["xx"] = np.ones(n_events)
     data_sets["yy"] = np.ones(n_events)
     data_sets["zz"] = np.ones(n_events)
-    data_sets["event_ids"] = np.arange(n_events)
+    # the shower energy
+    data_sets["shower_energies"] = np.ones(n_events) * 1 * units.EeV
```

Comment:
Why are showers initialized with 1EeV?

Reply:
this is a valid point. We probably shouldn't suggest to set shower energies manually at all? Is it less error prone, to provide external functions for generating an energy spectrum from the attributes?... like a powerlaw between Emin, Emax (ok we don't store an index (yet)). (don't know if this unnecessarily overcomplicate things for us, the optical telescopes typically use powerlaws between emin/max in the generator)

Reply:
this skeleton is intended for non standard use cases. So manually setting shower energies is something that people need to know how to do